### ISSUE TITLE
Incompatibility with Split testing Divi builder feature

### ISSUE THREAD

**Issue by piotrbak** [wp-media/wp-rocket#1613]
When the Split testing feature is enabled on the Divi Builder element, the mentioned element disappears from the page source when WP Rocket is active.

Disabling WP Rockets' features doesn't help, nor setting 'Never cache this page' in the page editor.

The only difference in the page source (except the lack of the whole section) can be seen here:
https://jmp.sh/XHTWktB

To reproduce:
1. Edit page using Divi Builder
2. Right click on the section
3. Enable Split Testing
4. Save post
5. Make sure WP Rocket is enabled

**Comment by GeekPress** [comment]
Some infos from Divi about how it works:

"The Split Testing is using JavaScript and Cookies in order to display different variations and collect the statistic. Code is also based on PHP in case if you have enabled OpCache or similiar feature. Different URLs are not used since you can't split test different pages."

from https://secure.helpscout.net/conversation/816154318/102348/

As their split feature is using PHP, I don't think we can be compatible.

**Comment by arunbasillal** [comment]
I was able to replicate this. Even when the page is excluded from cache and even with .htaccess rules removed and even setting `WP_CACHE` to `false`, didn't help. 

Turns out Divi builder has a function `et_pb_detect_cache_plugins` (in `/wp-content/themes/Divi/includes/builder/core.php` ) that checks for individual cache plugins. 

And it doesn't seem to be just WP Rocket - https://wordpress.org/support/topic/conflict-with-divi-split-testing/

I think the best solution right now is to wait and see if the Divi team can come up with improvements. They seem to attempt to accommodate for cache plugins. 

**Idea:**

The function `et_pb_detect_cache_plugins` is pluggable and can be over-ridden I think since it checks for the function during it's declaration. 

So in theory we could over ride it on pages that are excluded from cache and make WP Rocket compatible on pages excluded from cache in the time being if needed. Unless a user wants to do split test on 100% of the pages, they will still be able to use WP Rocket and just exclude the pages with split test from cache.
